src/util/records.py

```diff
@@ -808,8 +808,10 @@ def rename_xmlid(cr, old, new, noupdate=None, on_collision="fail"):
                 # Adapting t-call and t-name references in views
                 search_pattern = r"""\yt-(call|name)=(["']){}\2""".format(re.escape(old))
                 replace_pattern = r"t-\1=\2{}\2".format(new)
-                if version_gte("16.0"):
-                    arch_col = get_value_or_en_translation(cr, "ir_ui_view", "arch_db")
+                arch_col = get_value_or_en_translation(
+                    cr, "ir_ui_view", "arch_db" if column_exists(cr, "ir_ui_view", "arch_db") else "arch"
+                )
+                if version_gte("16.0") and "->>'en_US'" in str(arch_col):
```

Comment:
Actually the former check `version_gte("16.0")` should be `util.column_type(cr, "ir_ui_view", "arch_db") == "jsonb"`